dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,15 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory() const
+    {
+        IPropertyTree *history = root->queryPropTree("History");
+        //It can be null if the file created with an older version
+        if (!history)
+            history = root->setPropTree("History",createPTree("History")); // takes ownership
```

Comment:
So legal, but logically this object (root) is a member of class and this method is altering it.
Anyway, interesting more than anything, I wouldn't change, I'm sure there are plenty of other examples in our code base.
Also interestingly looks like they may introduce a mechanism to protect member pointer objects : http://en.cppreference.com/w/cpp/experimental/propagate_const